From 61b019b7a52750f965a8e93f1f3f0678d9e7c463 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:54:32 -0400
Subject: [fs] ext4: silence warnings in extent status tree debugging code

Message-id: <1414058125-4183-119-git-send-email-lczerner@redhat.com>
Patchwork-id: 98238
O-Subject: [RHEL 7.1 PATCH v2 118/171] ext4: silence warnings in extent status tree debugging code
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Eric Whitney <enwlinux@gmail.com>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit ce140cdd9c171dc75cfdcfee2b8708c508f5daf6

 Adjust the conversion specifications in a few optionally compiled debug
 messages to match the return type of ext4_es_status().  Also, make a
 couple of minor grammatical message edits while we're at it.

 Signed-off-by: Eric Whitney <enwlinux@gmail.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/extents_status.c b/fs/ext4/extents_status.c
index d9a61e1..6f98747 100644
--- a/fs/ext4/extents_status.c
+++ b/fs/ext4/extents_status.c
@@ -184,7 +184,7 @@ static void ext4_es_print_tree(struct inode *inode)
  while (node) {
   struct extent_status *es;
   es = rb_entry(node, struct extent_status, rb_node);
-  printk(KERN_DEBUG " [%u/%u) %llu %llx",
+  printk(KERN_DEBUG " [%u/%u) %llu %x",
          es->es_lblk, es->es_len,
          ext4_es_pblock(es), ext4_es_status(es));
   node = rb_next(node);
@@ -445,8 +445,8 @@ static void ext4_es_insert_extent_ext_check(struct inode *inode,
     pr_warn("ES insert assertion failed for "
      "inode: %lu we can find an extent "
      "at block [%d/%d/%llu/%c], but we "
-     "want to add an delayed/hole extent "
-     "[%d/%d/%llu/%llx]\n",
+     "want to add a delayed/hole extent "
+     "[%d/%d/%llu/%x]\n",
      inode->i_ino, ee_block, ee_len,
      ee_start, ee_status ? 'u' : 'w',
      es->es_lblk, es->es_len,
@@ -486,8 +486,8 @@ static void ext4_es_insert_extent_ext_check(struct inode *inode,
   if (!ext4_es_is_delayed(es) && !ext4_es_is_hole(es)) {
    pr_warn("ES insert assertion failed for inode: %lu "
     "can't find an extent at block %d but we want "
-    "to add an written/unwritten extent "
-    "[%d/%d/%llu/%llx]\n", inode->i_ino,
+    "to add a written/unwritten extent "
+    "[%d/%d/%llu/%x]\n", inode->i_ino,
     es->es_lblk, es->es_lblk, es->es_len,
     ext4_es_pblock(es), ext4_es_status(es));
   }
@@ -524,7 +524,7 @@ static void ext4_es_insert_extent_ind_check(struct inode *inode,
     */
    pr_warn("ES insert assertion failed for inode: %lu "
     "We can find blocks but we want to add a "
-    "delayed/hole extent [%d/%d/%llu/%llx]\n",
+    "delayed/hole extent [%d/%d/%llu/%x]\n",
     inode->i_ino, es->es_lblk, es->es_len,
     ext4_es_pblock(es), ext4_es_status(es));
    return;
@@ -554,7 +554,7 @@ static void ext4_es_insert_extent_ind_check(struct inode *inode,
   if (ext4_es_is_written(es)) {
    pr_warn("ES insert assertion failed for inode: %lu "
     "We can't find the block but we want to add "
-    "an written extent [%d/%d/%llu/%llx]\n",
+    "a written extent [%d/%d/%llu/%x]\n",
     inode->i_ino, es->es_lblk, es->es_len,
     ext4_es_pblock(es), ext4_es_status(es));
    return;
-- 
1.7.1